### PR TITLE
Add reduction related content to documents

### PR DESCRIPTION
**Pull request by CongMa13**
https://advanced-micro-devices-demo--248.com.readthedocs.build/projects/hipTensor/en/248/

### CODE REVIEW

**Review by cgmillette — DISMISSED** [review]

**Comment by CongMa13** [comment]
Hi @ROCm/rocm-documentation , could anyone help me to review this PR?

File: docs/what-is-hiptensor.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ What is hipTensor?
 
 hipTensor is a work-in-progress (WIP) high-performance HIP library for tensor primitives. It is AMD's C++ library for accelerating tensor primitives which can leverage specialized GPU matrix cores on AMD's latest discrete GPUs. hipTensor is currently powered by the composable kernel library. The API is designed to be portable with the Nvidia cuTensor library, allowing those users to easily migrate to the AMD platform.
```
Comment (lpaoletti) [inline docs/what-is-hiptensor.rst]: I think this topic might need to be removed. 
It looks like the same content is on the landing page (index.rst)

Comment (lpaoletti) [reply docs/what-is-hiptensor.rst]: Don't see it here:
image

File: docs/index.rst [hunk 2]
```diff
@@ -10,7 +10,7 @@ hipTensor documentation
...
+The hipTensor API offers functionality for defining tensor data objects and currently supports contraction, permutation and reduction operations on the tensor objects. Users may also control several available logging options. Under the hood, the hipTensor library is bundled with multitude of GPU kernels which are automatically selected and invoked to solve problems as defined by input parameters to the API. As hipTensor is currently a WIP, future tensor operation support may include reductions, element-wise operations and caching of selection instances.
 
 Supporting host code is required for GPU device and memory management. The hipTensor code samples and tests provided are built and launched via the Heterogeneous-Compute Interface for Portability (HIP) ecosystem within ROCm.
```
Comment (lpaoletti) [inline docs/index.rst:16]: Please change the following line to:
The code is open and hosted at 'Github...

File: docs/index.rst [hunk 3]
```diff
@@ -10,7 +10,7 @@ hipTensor documentation
 
 hipTensor is a work-in-progress (WIP) high-performance HIP library for tensor primitives. It is AMD's C++ library for accelerating tensor primitives which can leverage specialized GPU matrix cores on AMD's latest discrete GPUs. hipTensor is currently powered by the composable kernel library. The API is designed to be portable with the Nvidia cuTensor library, allowing those users to easily migrate to the AMD platform.
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
hipTensor is a work-in-progress (WIP) high-performance Heterogeneous-Compute Interface for Portability (HIP) library for tensor primitives. It’s AMD’s C++ library for accelerating tensor primitives, which can leverage specialized GPU matrix cores on AMD’s latest discrete GPUs. hipTensor is currently powered by the composable kernel (CK) library backend. The API is designed to be portable with the NVIDIA CUDA cuTensor library, allowing those users to easily migrate to the AMD platform.
```

File: docs/index.rst [hunk 4]
```diff
@@ -10,7 +10,7 @@ hipTensor documentation
...
 
-The hipTensor API offers functionality for defining tensor data objects and currently supports contraction and permutation operations on the tensor objects. Users may also control several available logging options. Under the hood, the hipTensor library is bundled with multitude of GPU kernels which are automatically selected and invoked to solve problems as defined by input parameters to the API. As hipTensor is currently a WIP, future tensor operation support may include reductions, element-wise operations and caching of selection instances.
+The hipTensor API offers functionality for defining tensor data objects and currently supports contraction, permutation and reduction operations on the tensor objects. Users may also control several available logging options. Under the hood, the hipTensor library is bundled with multitude of GPU kernels which are automatically selected and invoked to solve problems as defined by input parameters to the API. As hipTensor is currently a WIP, future tensor operation support may include reductions, element-wise operations and caching of selection instances.
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
The hipTensor API offers functionality for defining tensor data objects and currently supports contraction, permutation and reduction operations on the tensor objects. Users may also control several available logging options. The hipTensor library is bundled with GPU kernel instances, which are automatically selected and invoked to solve problems as defined by the API input parameters. As hipTensor is a WIP, future tensor operation support might include additional element-wise operations and caching of selection instances.
```

File: docs/index.rst [hunk 5]
```diff
@@ -10,7 +10,7 @@ hipTensor documentation
...
 
 Supporting host code is required for GPU device and memory management. The hipTensor code samples and tests provided are built and launched via the Heterogeneous-Compute Interface for Portability (HIP) ecosystem within ROCm.
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
Supporting host code is required for GPU device and memory management. The hipTensor code samples and tests provided are built and launched via the HIP ecosystem within ROCm.
```

File: docs/conceptual/programmers-guide.rst [hunk 6]
```diff
@@ -34,10 +34,10 @@ The outer API layer serves as a functional interface for the user to define tens
 between the API objects and the desired functionality. The solution layer encompasses the translation of the input problem parameters into solution candidates, candidate selection, resource management and logging.
```
Comment (lpaoletti) [inline docs/conceptual/programmers-guide.rst:34]: Change title to:
hipTensor programmer's guide

On line 3, add the keyword "cuTensor"

File: docs/conceptual/programmers-guide.rst [hunk 7]
```diff
@@ -54,6 +54,11 @@ Tensor permutation
 
 Tensor permutation is essentially the re-ordering of the stride indices such that the data dimensional locality relationships are changed.
 
+Tensor reduction
+^^^^^^^^^^^^^^^^^^^
+
+Tensor reduction refers to the process of obtaining a lower-dimensional tensor from a higher-dimensional one by performing some kind of operation on the original tensor.
```
Comment (lpaoletti) [inline docs/conceptual/programmers-guide.rst]:
```suggestion
Tensor reduction refers to obtaining a lower-dimensional tensor from a higher-dimensional one by performing some operation on the original tensor.
```

**Review by lpaoletti — COMMENTED** [review]
Please let me know if you want to discuss removing the what-is-hiptensor.rst file

File: docs/conceptual/programmers-guide.rst (hunk 6 above)
Comment (CongMa13) [reply docs/conceptual/programmers-guide.rst:34]: > Change title to: hipTensor programmer's guide
> 
> On line 3, add the keyword "cuTensor"

Hi @lpaoletti, just wondering if it is common practice to include cuda library name in rocm document?

Comment (lpaoletti) [reply docs/conceptual/programmers-guide.rst:34]: @CongMa13 - yes, we sometimes note the alternative/parallel library 
My suggestion is to add it to the keywords (line 3). This is used to build the search indexing. 

By the way, the introduction also includes a reference to CUDA:
image

Comment (CongMa13) [reply docs/conceptual/programmers-guide.rst:34]: Perfect. Thanks for the explanation. Just want to make sure that everything is complied with the documentation policies.

File: docs/index.rst (hunk 4 above)
Comment (cgmillette) [reply docs/index.rst]: Reductions are currently being added! :)

File: docs/what-is-hiptensor.rst (hunk 1 above)
Comment (cgmillette) [reply docs/what-is-hiptensor.rst]: @CongMa13 Good to remove

Comment (CongMa13) [reply docs/what-is-hiptensor.rst]: removed

**Comment by CongMa13** [comment]
> Please let me know if you want to discuss removing the what-is-hiptensor.rst file

Hi @lpaoletti , thanks for reviewing the PR. The file has been removed. Could you review it again?

**Comment by CongMa13** [comment]
Hi @lpaoletti, the documents have been modified according to your suggestion. If there are no issues, could you approve it? Thanks. 😃

**Review by lpaoletti — APPROVED** [review]

**Review by cgmillette — APPROVED** [review]

**Pull request merged**